### PR TITLE
Create temp dir for standalone model file

### PR DESCRIPTION
**Pull request by srchase**
This PR updates the how the language server is initialized to handle editing a standalone smithy model file. When a client is editing a single model file, a workspace may not have been created yet. This PR checks if the workspace is present in the server initialization params and creates a temporary directory to use in its absence.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java [hunk 1]
```diff
@@ -77,6 +77,18 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
       LspLog.println("Workspace root was null");
     }
 
+    if (params.getWorkspaceFolders() == null) {
+      try {
+        File temp = Files.createTempDirectory("tmp-smithy-language-server-workspace").toFile();
+        System.out.println("Created temporary workspace root: " + temp);
+        temp.deleteOnExit();
+        WorkspaceFolder workspaceFolder = new WorkspaceFolder(temp.toURI().toString());
+        params.setWorkspaceFolders(ImmutableList.of(workspaceFolder));
```
Comment (DavidOgunsAWS) [inline src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java:86]: An editor could add/remove workspace folders at any time and I'm not sure if lsp4j will defensively copy this list. If it doesn't and tries to modify the same list that'll be a sure bug (exception thrown) later. If it's too hard to verify should be happening here I'd be more comfortable with a mutable list instead.

File: src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java [hunk 2]
```diff
@@ -77,6 +77,18 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
       LspLog.println("Workspace root was null");
     }
 
+    if (params.getWorkspaceFolders() == null) {
+      try {
+        File temp = Files.createTempDirectory("tmp-smithy-language-server-workspace").toFile();
+        System.out.println("Created temporary workspace root: " + temp);
```
Comment (DavidOgunsAWS) [inline src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java:83]: Should this output go somewhere other than stdout directly?

**Review by DavidOgunsAWS — COMMENTED** [review]
(ignore, moved comment)

File: src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java (hunk 1 above)
Comment (srchase) [reply src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java]: Switched to `ListUtils.of` in https://github.com/awslabs/smithy-language-server/pull/54/commits/033ddbafabccfaea4a295f273e9d0e89811b9c77.

**Review by DavidOgunsAWS — APPROVED** [review]

**Pull request merged**